compiler/rustc_trait_selection/src/traits/error_reporting/on_unimplemented.rs

```diff
@@ -184,14 +184,7 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
             flags.push((sym::cause, Some("MainFunctionType".to_string())));
         }
 
-        if let Some(kind) = self.tcx.fn_trait_kind_from_def_id(trait_ref.def_id)
-            && let ty::Tuple(args) = trait_ref.args.type_at(1).kind()
-        {
-            let args = args.iter().map(|ty| ty.to_string()).collect::<Vec<_>>().join(", ");
-            flags.push((sym::Trait, Some(format!("{}({args})", kind.as_str()))));
-        } else {
-            flags.push((sym::Trait, Some(trait_ref.print_only_trait_path().to_string())));
-        }
+        flags.push((sym::Trait, Some(trait_ref.print_trait_sugared().to_string())));
```

Comment:
Have you checked all the `rustc_on_unimplemented` that reference `Trait` in their filters? Probably just the ones in the definitions for `Fn`, `FnOnce` and `FnMut` in particular.

Reply:
Yes, this PR should be a no-op, since it just generalizes the behavior that I replaced here.

I'm also the one that introduced `sym::Trait`, so I know that it's probably not gonna affect anything.

Reply:
Ah, yeah, didn't realize that. Awesome.